<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>FirstRunDialog</class>
 <widget class="QDialog" name="FirstRunDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>683</width>
    <height>480</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>First Run - GP Developer</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <property name="spacing">
    <number>0</number>
   </property>
   <property name="margin">
    <number>0</number>
   </property>
   <item>
    <widget class="QWidget" name="titleWidget" native="true">
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <item>
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string/>
        </property>
        <property name="pixmap">
         <pixmap resource="icons.qrc">:/icons/comment.png</pixmap>
        </property>
        <property name="scaledContents">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="title_2">
        <property name="styleSheet">
         <string notr="true">QLabel { font-size: 16px; }</string>
        </property>
        <property name="text">
         <string>Welcome to GP Developer</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>159</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QWidget" name="dialogMainWidget" native="true">
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="Developer::GraphWidget" name="graphWidget">
        <property name="minimumSize">
         <size>
          <width>340</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>340</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="styleSheet">
         <string notr="true">QGraphicsView { border-right: 1px solid #ccc; }</string>
        </property>
        <property name="frameShape">
         <enum>QFrame::NoFrame</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QStackedWidget" name="stackedWidget">
        <widget class="QWidget" name="introduction">
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <widget class="QLabel" name="label">
            <property name="text">
             <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Sans'; font-size:10pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:12pt;&quot;&gt;Introduction&lt;/span&gt;&lt;/p&gt;
&lt;p align=&quot;justify&quot; style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Please take your time to follow this quick introductory guide to how graphs are created and manipulated within the GP Developer.&lt;/p&gt;
&lt;p align=&quot;justify&quot; style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;If you have used GP Developer before then feel free to skip this using the button in the bottom left, and if you ever need to remind yourself on the mechanics of this application then this dialog can be brought up at any time from the &amp;quot;Help&amp;quot; menu.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignJustify|Qt::AlignVCenter</set>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="addingNodes">
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:12pt;&quot;&gt;Adding Nodes&lt;/span&gt;&lt;/p&gt;&lt;p&gt;To create a node you can double click the left mouse button (&lt;img src=&quot;:/icons/small_mouse_select_left.png&quot;/&gt;) anywhere in open space of the graph. Do this now to create at least three nodes in the graph widget to the left.&lt;/p&gt;&lt;p&gt;Once a node has been added to the graph it can be moved by clicking the left mouse button over the desired node and then dragging until the node is in the desired position.&lt;/p&gt;&lt;p&gt;You do not have to lay out the nodes yourself however, there are automated layout mechanisms available which will be discussed later.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignJustify|Qt::AlignVCenter</set>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <item>
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>Nodes created:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="nodeCount">
              <property name="text">
               <string>0</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="nodeCountIndicator">
              <property name="text">
               <string/>
              </property>
              <property name="pixmap">
               <pixmap resource="icons.qrc">:/icons/small_cross.png</pixmap>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <spacer name="verticalSpacer_2">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>124</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="addingEdges">
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <widget class="QLabel" name="label_6">
            <property name="frameShadow">
             <enum>QFrame::Plain</enum>
            </property>
            <property name="text">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:12pt;&quot;&gt;Adding Edges&lt;/span&gt;&lt;/p&gt;&lt;p&gt;Edges between nodes can be added by clicking the right mouse button (&lt;img src=&quot;:/icons/small_mouse_select_right.png&quot;/&gt;) over an existing node and dragging until your cursor is over another node. When you release the right mouse button an edge is added from the first node to the second.&lt;/p&gt;&lt;p&gt;Loop edges (edges from a node to itself) can be added with a single right mouse button click over the desired node.&lt;/p&gt;&lt;p&gt;Add at least three edges to the graph now.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignJustify|Qt::AlignVCenter</set>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_5">
            <item>
             <widget class="QLabel" name="label_7">
              <property name="text">
               <string>Edges added:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="edgeCount">
              <property name="text">
               <string>0</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="edgeCountIndicator">
              <property name="text">
               <string/>
              </property>
              <property name="pixmap">
               <pixmap resource="icons.qrc">:/icons/small_cross.png</pixmap>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <spacer name="verticalSpacer_3">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="editingGraphElements">
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <item>
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:12pt;&quot;&gt;Editing Graph Elements&lt;/span&gt;&lt;/p&gt;&lt;p&gt;Elements which have been added to the graph can be edited by using the left mouse button (&lt;img src=&quot;:/icons/small_mouse_select_left.png&quot;/&gt;) to double click on the element you want to change. A dialog will appear which allows you to make any necessary changes to this element, such as changing its identifier or label.&lt;/p&gt;&lt;p&gt;Try editing the elements you have created to see how they can be modified.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer_4">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>218</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="selectingGraphElements">
         <layout class="QVBoxLayout" name="verticalLayout_7">
          <item>
           <widget class="QLabel" name="label_11">
            <property name="text">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:12pt;&quot;&gt;Selecting Elements&lt;/span&gt;&lt;/p&gt;&lt;p&gt;Individual elements (both edges and nodes) can be selected with a single left mouse button (&lt;img src=&quot;:/icons/small_mouse_select_left.png&quot;/&gt;) click on the element you wish to select.&lt;/p&gt;&lt;p&gt;If you wish to select many elements at once you can click the left mouse button in open space and then drag a rectangle from that point. Any elements which overlap with the area indicated will be selected.&lt;/p&gt;&lt;p&gt;In addition it is possible to select every element in the graph using the keyboard shortcut Ctrl+A.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer_6">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>176</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="deletingGraphElements">
         <layout class="QVBoxLayout" name="verticalLayout_6">
          <item>
           <widget class="QLabel" name="label_8">
            <property name="text">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:12pt;&quot;&gt;Deleting Elements&lt;/span&gt;&lt;/p&gt;&lt;p&gt;When a single element or a group of elements are selected they can be deleted by pressing the &amp;quot;Delete&amp;quot; key on your keyboard.&lt;/p&gt;&lt;p&gt;Remove three nodes and edges now, create more if there are not enough to delete.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_7">
            <item>
             <widget class="QLabel" name="label_9">
              <property name="text">
               <string>Nodes Deleted:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="nodeDeleteCount">
              <property name="text">
               <string>0</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="nodeDeleteCountIndicator">
              <property name="text">
               <string/>
              </property>
              <property name="pixmap">
               <pixmap resource="icons.qrc">:/icons/small_cross.png</pixmap>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_6">
            <item>
             <widget class="QLabel" name="label_10">
              <property name="text">
               <string>Edges Deleted:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="edgeDeleteCount">
              <property name="text">
               <string>0</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="edgeDeleteCountIndicator">
              <property name="text">
               <string/>
              </property>
              <property name="pixmap">
               <pixmap resource="icons.qrc">:/icons/small_cross.png</pixmap>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <spacer name="verticalSpacer_5">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>222</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="navigatingGraphs">
         <layout class="QVBoxLayout" name="verticalLayout_8">
          <item>
           <widget class="QLabel" name="label_12">
            <property name="text">
             <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Sans'; font-size:10pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:12pt;&quot;&gt;Navigating Large Graphs&lt;/span&gt;&lt;/p&gt;
&lt;p align=&quot;justify&quot; style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;Large graphs may not fit easily into the available graph widget. In those cases the widget allows the user to pan around and zoom as necessary.&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:12pt;&quot;&gt;Moving Around&lt;/span&gt;&lt;/p&gt;
&lt;p align=&quot;justify&quot; style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;To move around the graph you can use the scrollbars shown at the bottom and at the right (as necessary). In addition when the &amp;quot;Shift&amp;quot; key is held the widget allows the user to pan around by clicking the left mouse button (&lt;/span&gt;&lt;img src=&quot;:/icons/small_mouse_select_left.png&quot; /&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;) and moving the mouse.&lt;/span&gt;&lt;/p&gt;
&lt;p align=&quot;justify&quot; style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:12pt;&quot;&gt;Zooming&lt;/span&gt;&lt;/p&gt;
&lt;p align=&quot;justify&quot; style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;Zooming of the view can be achieved by rolling the mouse wheel (&lt;/span&gt;&lt;img src=&quot;:/icons/small_mouse_select_scroll.png&quot; /&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;) or by pressing Ctrl and &amp;quot;+&amp;quot; to zoom in and Ctrl and &amp;quot;-&amp;quot; to zoom out. You can reset the view to a 1:1 scale at any time by pressing Ctrl+0 (zero).&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer_7">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>21</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QWidget" name="buttons" native="true">
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QPushButton" name="skipButton">
        <property name="text">
         <string>Skip Introduction</string>
        </property>
        <property name="icon">
         <iconset resource="icons.qrc">
          <normaloff>:/icons/small_door_out.png</normaloff>:/icons/small_door_out.png</iconset>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>272</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="backButton">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Back</string>
        </property>
        <property name="icon">
        <iconset resource="icons.qrc">
          <normaloff>:/icons/small_arrow_left.png</normaloff>:/icons/small_arrow_left.png</iconset>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="nextButton">
        <property name="text">
         <string>Next</string>
        </property>
        <property name="icon">
        <iconset resource="icons.qrc">
          <normaloff>:/icons/small_arrow_right.png</normaloff>:/icons/small_arrow_right.png</iconset>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="beginButton">
        <property name="text">
         <string>End</string>
        </property>
        <property name="icon">
        <iconset resource="icons.qrc">
          <normaloff>:/icons/small_tick.png</normaloff>:/icons/small_tick.png</iconset>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Developer::GraphWidget</class>
   <extends>QGraphicsView</extends>
   <header>graphview/graphwidget.hpp</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>skipButton</sender>
   <signal>pressed()</signal>
   <receiver>FirstRunDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>84</x>
     <y>466</y>
    </hint>
    <hint type="destinationlabel">
     <x>85</x>
     <y>524</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>beginButton</sender>
   <signal>pressed()</signal>
   <receiver>FirstRunDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>609</x>
     <y>470</y>
    </hint>
    <hint type="destinationlabel">
     <x>610</x>
     <y>514</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>nextButton</sender>
   <signal>pressed()</signal>
   <receiver>FirstRunDialog</receiver>
   <slot>next()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>530</x>
     <y>462</y>
    </hint>
    <hint type="destinationlabel">
     <x>535</x>
     <y>518</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>backButton</sender>
   <signal>pressed()</signal>
   <receiver>FirstRunDialog</receiver>
   <slot>back()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>450</x>
     <y>463</y>
    </hint>
    <hint type="destinationlabel">
     <x>442</x>
     <y>554</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>back()</slot>
  <slot>next()</slot>
 </slots>
</ui>
